hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/CreateHoodieTableAsSelectCommand.scala

```diff
@@ -73,9 +75,10 @@ case class CreateHoodieTableAsSelectCommand(
 
     // Execute the insert query
     try {
+      val tblProperties = table.storage.properties ++ table.properties
```

Comment:
have seen this needed repeated. can we consider making a `HoodieCatalogTable` to encapsulate this and other hudi specific logic inside, e.g. validation, options transform, etc

Reply:
good idea.